framework/src/markers/ReporterPointMarker.C

```diff
@@ -0,0 +1,74 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ReporterPointMarker.h"
+
+registerMooseObject("MooseApp", ReporterPointMarker);
+
+InputParameters
+ReporterPointMarker::validParams()
+{
+  InputParameters params = Marker::validParams();
+  params.addClassDescription("Marks the region inside or empty if it contains a reporter defined "
+                             "point for refinement or coarsening.");
+  params.addRequiredParam<ReporterName>("x_coord_name", "reporter x-coordinate name");
+  params.addRequiredParam<ReporterName>("y_coord_name", "reporter y-coordinate name");
+  params.addRequiredParam<ReporterName>("z_coord_name", "reporter z-coordinate name");
+  MooseEnum marker_states = Marker::markerStates();
+  params.addRequiredParam<MooseEnum>(
+      "inside", marker_states, "How to mark elements containing a point");
+  params.addRequiredParam<MooseEnum>(
+      "empty", marker_states, "How to mark elements not containing a point");
+  return params;
+}
+
+ReporterPointMarker::ReporterPointMarker(const InputParameters & parameters)
+  : Marker(parameters),
+    ReporterInterface(this),
+    _inside(parameters.get<MooseEnum>("inside").getEnum<MarkerValue>()),
+    _empty(parameters.get<MooseEnum>("empty").getEnum<MarkerValue>()),
+    _x_coord(getReporterValue<std::vector<Real>>("x_coord_name", REPORTER_MODE_REPLICATED)),
+    _y_coord(getReporterValue<std::vector<Real>>("y_coord_name", REPORTER_MODE_REPLICATED)),
+    _z_coord(getReporterValue<std::vector<Real>>("z_coord_name", REPORTER_MODE_REPLICATED))
```

Comment:
wit not make it an `std::vector<Point>`?

Reply:
The reporter values will change.  I'm using this in a multiapp, so at the start, the reporter is empty and then gets filled after the multiapp executes.

Reply:
?

Can't the reporter provide a vector of points?

Reply:
I'm using the constant reporter and it won't hold points.  I think I would need to create a new reporter that had points.

Reply:
It should be really easy to add `Point` to the ConstantReporter options if you want to go that route. Should be two lines of code.